### PR TITLE
misc: logger, mode permission and storages

### PR DESCRIPTION
**Pull request by ZachCheung**
* logger f

```go
a := "world"
logger.Print(a)
logger.Printf("hello %v!", a)
logger.Println(a)
logger.Debug(a)
logger.Debugf("hello %v!", a)
logger.Info(a)
logger.Infof("hello %v!", a)
logger.Warn(a)
logger.Warnf("hello %v!", a)
logger.Error(a)
logger.Errorf("hello %v!", a)
logger.Fatalf("hello %v!", a)
```
logger

* more logs
* change cycler config file permission
* fixed `model.Storages` `store_with` bug
* add `GOBACKUP_DIR` env

### CODE REVIEW

**Comment by ZachCheung** [comment]
Use temp dir as default `workdir` instead of a fixed `/tmp/gobackup`

### Before

```
$ sudo ./gobackup perform
2022/12/03 01:30:48 [Modal: demo] WorkDir: /tmp/gobackup/1670059848802701471/demo
2022/12/03 01:30:48 [Database] => database | postgresql : postgresql
2022/12/03 01:30:48 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 01:30:49 [PostgreSQL] dump path: /tmp/gobackup/1670059848802701471/demo/postgresql/postgresql/demo.sql
2022/12/03 01:30:49 [Database]
2022/12/03 01:30:49 [Compressor] => Compress | xz
2022/12/03 01:30:49 [Compressor] -> /tmp/gobackup/1670059848802701471/2022.12.03.01.30.49.tar.xz
2022/12/03 01:30:49 [Storage] => Storage | local
2022/12/03 01:30:49 [Local] Store successed /tmp/demo1
2022/12/03 01:30:49 [Model] Cleanup temp: /tmp/gobackup/1670059848802701471/

$ ./gobackup perform
2022/12/03 01:30:57 [Modal: demo] WorkDir: /tmp/gobackup/1670059857936056064/demo
2022/12/03 01:30:57 [Database] => database | postgresql : postgresql
2022/12/03 01:30:57 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 01:30:58 [debug] /usr/bin/pg_dump --host=localhost --port=5432 --username=postgres demo -f /tmp/gobackup/1670059857936056064/demo/postgresql/postgresql/demo.sql
2022/12/03 01:30:58 [Modal: demo] pg_dump: error: could not open output file "/tmp/gobackup/1670059857936056064/demo/postgresql/postgresql/demo.sql": No such file or directory

2022/12/03 01:30:58 [Model] Cleanup temp: /tmp/gobackup/1670059857936056064/
```

### After

```
$ mkdir /tmp/demo
$ sudo ./gobackup perform
2022/12/03 02:56:23 [Modal: demo] WorkDir: /tmp/gobackup4023818121/1670064983565757453/demo
2022/12/03 02:56:23 [Database] => database | postgresql: postgresql
2022/12/03 02:56:23 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 02:56:23 [PostgreSQL] dump path: /tmp/gobackup4023818121/1670064983565757453/demo/postgresql/postgresql/demo.sql
2022/12/03 02:56:23 [Database]
2022/12/03 02:56:23 [Compressor] => Compress | xz
2022/12/03 02:56:23 [Compressor] -> /tmp/gobackup4023818121/1670064983565757453/2022.12.03.02.56.23.tar.xz
2022/12/03 02:56:23 [Storage] => Storage | local
2022/12/03 02:56:23 [Local] Store succeeded /tmp/demo
2022/12/03 02:56:23 [Model] Cleanup temp: /tmp/gobackup4023818121/

$ ./gobackup perform
2022/12/03 02:56:26 [Modal: demo] WorkDir: /tmp/gobackup2923488212/1670064986188901362/demo
2022/12/03 02:56:26 [Database] => database | postgresql: postgresql
2022/12/03 02:56:26 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 02:56:26 [PostgreSQL] dump path: /tmp/gobackup2923488212/1670064986188901362/demo/postgresql/postgresql/demo.sql
2022/12/03 02:56:26 [Database]
2022/12/03 02:56:26 [Compressor] => Compress | xz
2022/12/03 02:56:26 [Compressor] -> /tmp/gobackup2923488212/1670064986188901362/2022.12.03.02.56.26.tar.xz
2022/12/03 02:56:26 [Storage] => Storage | local
2022/12/03 02:56:26 [Local] Store succeeded /tmp/demo
2022/12/03 02:56:26 [Model] Cleanup temp: /tmp/gobackup2923488212/
```

**Comment by ZachCheung** [comment]
Added `prestart`, `poststart`, `always_poststart` configs for database.

Scenario:
1. `SELECT pg_wal_replay_resume();` to pause replay in PostgreSQL replica
2. `pg_dump`
3. `SELECT select pg_wal_replay_resume();` to resume replay in PostgreSQL replica no matter `pg_dump` succeeds or not

```
2022/12/03 04:09:31 [Model: demo] WorkDir: /tmp/gobackup1781575701/1670069371448053572/demo
2022/12/03 04:09:31 [Database] => database | postgresql: postgresql
2022/12/03 04:09:31 [Database] Run dump prestart command
2022/12/03 04:09:31 [Database] Dump prestart command succeeded
2022/12/03 04:09:31 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 04:09:31 [debug]  /usr/bin/pg_dump --host=localhost --port=5432 --username=postgres demo -f /tmp/gobackup1781575701/1670069371448053572/demo/postgresql/postgresql/demo.sql
2022/12/03 04:09:31 [Database] Dump failed
2022/12/03 04:09:31 [Database] always_poststart is true, start to run post start command
2022/12/03 04:09:31 [Database] Run dump poststart command
2022/12/03 04:09:31 [Database] Dump poststart command succeeded
2022/12/03 04:09:31 [Model: demo] pg_dump: error: connection to server at "localhost" (::1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

2022/12/03 04:09:31 [Model] Cleanup temp: /tmp/gobackup1781575701/
```

**Comment by ZachCheung** [comment]
rename hooks

* `prestart` => `before`
* `poststart` => `after`
* `always_poststart` => `on_exit`: `always`, `success`, `failure`

#### before + after + `always` on_exit

```
2022/12/03 18:48:54 [Model: demo] WorkDir: /tmp/gobackup3821489654/1670122134570844525/demo
2022/12/03 18:48:54 [Database] => database | postgresql: postgresql
2022/12/03 18:48:54 [Database] Run dump before hooks
2022/12/03 18:48:54 [Database] Dump before hooks succeeded
2022/12/03 18:48:54 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 18:48:54 [debug]  /usr/bin/pg_dump --host=localhost --port=5432 --username=postgres demo -f /tmp/gobackup3821489654/1670122134570844525/demo/postgresql/postgresql/demo.sql
2022/12/03 18:48:54 [Database] Dump failed
2022/12/03 18:48:54 [Database] on_exit is always, start to run after hooks
2022/12/03 18:48:54 [Database] Run dump after hooks
2022/12/03 18:48:54 [Database] Dump after hooks succeeded
2022/12/03 18:48:54 [Model: demo] pg_dump: error: connection to server at "localhost" (::1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

2022/12/03 18:48:54 [Model] Cleanup temp: /tmp/gobackup3821489654/
```

#### before + after + `success` on_exit

```
2022/12/03 18:49:21 [Model: demo] WorkDir: /tmp/gobackup4250935195/1670122161319166282/demo
2022/12/03 18:49:21 [Database] => database | postgresql: postgresql
2022/12/03 18:49:21 [Database] Run dump before hooks
2022/12/03 18:49:21 [Database] Dump before hooks succeeded
2022/12/03 18:49:21 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 18:49:21 [debug]  /usr/bin/pg_dump --host=localhost --port=5432 --username=postgres demo -f /tmp/gobackup4250935195/1670122161319166282/demo/postgresql/postgresql/demo.sql
2022/12/03 18:49:21 [Database] Dump failed
2022/12/03 18:49:21 [Database] on_exit is success, skip run after hooks
2022/12/03 18:49:21 [Model: demo] pg_dump: error: connection to server at "localhost" (::1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

2022/12/03 18:49:21 [Model] Cleanup temp: /tmp/gobackup4250935195/
```

#### before + after + `failure` on_exit

```
2022/12/03 18:49:33 [Model: demo] WorkDir: /tmp/gobackup2759939241/1670122173977769211/demo
2022/12/03 18:49:33 [Database] => database | postgresql: postgresql
2022/12/03 18:49:33 [Database] Run dump before hooks
2022/12/03 18:49:33 [Database] Dump before hooks succeeded
2022/12/03 18:49:33 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 18:49:33 [debug]  /usr/bin/pg_dump --host=localhost --port=5432 --username=postgres demo -f /tmp/gobackup2759939241/1670122173977769211/demo/postgresql/postgresql/demo.sql
2022/12/03 18:49:33 [Database] Dump failed
2022/12/03 18:49:33 [Database] on_exit is failure, start to run after hooks
2022/12/03 18:49:33 [Database] Run dump after hooks
2022/12/03 18:49:33 [Database] Dump after hooks succeeded
2022/12/03 18:49:33 [Model: demo] pg_dump: error: connection to server at "localhost" (::1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

2022/12/03 18:49:33 [Model] Cleanup temp: /tmp/gobackup2759939241/
```

#### before + after + no on_exit

```
2022/12/03 18:49:39 [Model: demo] WorkDir: /tmp/gobackup1635565693/1670122179577122327/demo
2022/12/03 18:49:39 [Database] => database | postgresql: postgresql
2022/12/03 18:49:39 [Database] Run dump before hooks
2022/12/03 18:49:39 [Database] Dump before hooks succeeded
2022/12/03 18:49:39 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 18:49:39 [debug]  /usr/bin/pg_dump --host=localhost --port=5432 --username=postgres demo -f /tmp/gobackup1635565693/1670122179577122327/demo/postgresql/postgresql/demo.sql
2022/12/03 18:49:39 [Database] Dump failed
2022/12/03 18:49:39 [Model: demo] pg_dump: error: connection to server at "localhost" (::1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

2022/12/03 18:49:39 [Model] Cleanup temp: /tmp/gobackup1635565693/
```

#### before + no after

```
2022/12/03 18:49:56 [Model: demo] WorkDir: /tmp/gobackup1413645216/1670122196976374293/demo
2022/12/03 18:49:56 [Database] => database | postgresql: postgresql
2022/12/03 18:49:56 [Database] Run dump before hooks
2022/12/03 18:49:56 [Database] Dump before hooks succeeded
2022/12/03 18:49:56 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 18:49:56 [debug]  /usr/bin/pg_dump --host=localhost --port=5432 --username=postgres demo -f /tmp/gobackup1413645216/1670122196976374293/demo/postgresql/postgresql/demo.sql
2022/12/03 18:49:56 [Database] Dump failed
2022/12/03 18:49:56 [Model: demo] pg_dump: error: connection to server at "localhost" (::1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

2022/12/03 18:49:56 [Model] Cleanup temp: /tmp/gobackup1413645216/
```

#### no before + no after

```
2022/12/03 18:50:04 [Model: demo] WorkDir: /tmp/gobackup3975745782/1670122204116304516/demo
2022/12/03 18:50:04 [Database] => database | postgresql: postgresql
2022/12/03 18:50:04 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 18:50:04 [debug]  /usr/bin/pg_dump --host=localhost --port=5432 --username=postgres demo -f /tmp/gobackup3975745782/1670122204116304516/demo/postgresql/postgresql/demo.sql
2022/12/03 18:50:04 [Database] Dump failed
2022/12/03 18:50:04 [Model: demo] pg_dump: error: connection to server at "localhost" (::1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

2022/12/03 18:50:04 [Model] Cleanup temp: /tmp/gobackup3975745782/
```

#### no before + after + `always` on_exit

```
2022/12/03 18:55:29 [Model: demo] WorkDir: /tmp/gobackup1435367072/1670122529691722088/demo
2022/12/03 18:55:29 [Database] => database | postgresql: postgresql
2022/12/03 18:55:29 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/03 18:55:29 [debug]  /usr/bin/pg_dump --host=localhost --port=5432 --username=postgres demo -f /tmp/gobackup1435367072/1670122529691722088/demo/postgresql/postgresql/demo.sql
2022/12/03 18:55:29 [Database] Dump failed
2022/12/03 18:55:29 [Database] on_exit is always, start to run after hooks
2022/12/03 18:55:29 [Database] Run dump after hooks
2022/12/03 18:55:29 [Database] Dump after hooks succeeded
2022/12/03 18:55:29 [Model: demo] pg_dump: error: connection to server at "localhost" (::1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

2022/12/03 18:55:29 [Model] Cleanup temp: /tmp/gobackup1435367072/
```

File: database/base.go [hunk 1]
```diff
@@ -56,14 +61,62 @@ func runModel(model config.ModelConfig, dbConfig config.SubConfig) (err error) {
...
+		if err != nil {
+			return err
+		}
+		if _, err := helper.Exec(c[0], c[1:]...); err != nil {
+			return err
+		}
+		logger.Info("Dump before hooks succeeded")
+	}
+
+	after := dbConfig.Viper.GetString("after")
```
Comment (huacnlee) [inline database/base.go:79]: Rename to `after_script`

Comment (huacnlee) [reply database/base.go:79]: 这样后面，model 还可以增加一个 `script` 属性，用来执行 Bash Script

**Comment by ZachCheung** [comment]
add systemd-style `-` prefix to ignore hook exit

```
2022/12/04 19:43:25 [Model: demo] WorkDir: /tmp/gobackup2329085183/1670211805991203562/demo
2022/12/04 19:43:25 [Database] => database | postgresql: postgresql
2022/12/04 19:43:25 [Database] Run dump before_script
2022/12/04 19:43:25 [debug]  /tmp/prestart hello world 1
2022/12/04 19:43:25 [Database] Run dump before_script failed: , ignore it
2022/12/04 19:43:25 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/04 19:43:26 [PostgreSQL] dump path: /tmp/gobackup2329085183/1670211805991203562/demo/postgresql/postgresql/demo.sql
2022/12/04 19:43:26 [Database] Dump succeeded
2022/12/04 19:43:26 [Database] Run dump after_script
2022/12/04 19:43:26 [Model: demo] Run dump after_script failed: echoecho cannot be found
2022/12/04 19:43:26 [Model] Cleanup temp: /tmp/gobackup2329085183/
```

**Pull request merged**